### PR TITLE
Challenge 12 completed

### PR DESCRIPTION
**Pull request by CappaCoppa**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -294,4 +294,13 @@ describe('Comments Post request testing block', () => {
         })
     })
     
+});
+
+describe('Comments delete request testing block', () => {
+    test("Return an empty space after the deletion of the comment with specifi comment_id", () => {
+        return request(app).delete("/api/comments/2").expect(204).then((res) => {
+            console.log(res.body)
```
Comment (mallionaire) [inline __tests__/app.test.js]: remove console.logs

File: controllers/categories.controller.js [hunk 2]
```diff
@@ -12,4 +12,10 @@ const getCategories = (req, res, next) => {
     )
 }
 
-module.exports = {getCategories}
\ No newline at end of file
+exports.deleteComment = (req, res, next) => {
+    const id = req.params.comment_id
+    fetchDeletedComment(id).then((deletedComment) => {
+        console.log(deletedComment)
```
Comment (mallionaire) [inline controllers/categories.controller.js]: remove

**Review by mallionaire — CHANGES_REQUESTED** [review]
You'll need to add test cases for: 
 - 404 comment not found
 - 400 invalid comment id

**Review by mallionaire — CHANGES_REQUESTED** [review]
Still some console.logs in here, you'll need to remove them before it can be approved

File: __tests__/app.test.js [hunk 3]
```diff
@@ -294,4 +294,24 @@ describe('Comments Post request testing block', () => {
         })
     })
     
+});
+
+describe('Comments delete request testing block', () => {
+    test("Return an empty space after the deletion of the comment with specifi comment_id", () => {
+        return request(app).delete("/api/comments/1").expect(204).then((res) => {
+            expect(Object.keys(res.body).length).toEqual(0)
```
Comment (scar1377) [inline __tests__/app.test.js]: don't need the .then() block

File: controllers/categories.controller.js [hunk 4]
```diff
@@ -12,4 +12,15 @@ const getCategories = (req, res, next) => {
     )
 }
 
-module.exports = {getCategories}
\ No newline at end of file
+exports.deleteComment = (req, res, next) => {
+    const id = req.params.comment_id
+    fetchDeletedComment(id).then((deletedComment) => {
+        if(!deletedComment.length){
+            return Promise.reject({status: 404, msg : "comment_id in path but does not exist"})
```
Comment (scar1377) [inline controllers/categories.controller.js]: This part is better in the model. only leave.then in controller

File: models/categories.model.js [hunk 5]
```diff
@@ -12,4 +12,8 @@ const fetchCategories = () => {
     })
     }
 
-module.exports = {fetchCategories};
\ No newline at end of file
+exports.fetchDeletedComment = (id) => {
+    return db.query("DELETE FROM comments WHERE comment_id = $1 RETURNING *",[id]).then(({rows})=>{
```
Comment (scar1377) [inline models/categories.model.js]: can take a look at rowCount rather than rows. When It's 0, return Promise.reject

**Review by scar1377 — APPROVED** [review]
Good to merge

**Pull request merged**